### ISSUE TITLE
Design flaw for the link with the generated project

### ISSUE THREAD

**Issue by seb-elico** [OCA/business-requirement#115]
Long story short, it's possible to generate multiple projects from 1 BR but there's only 1 field to store the generated project ("Linked Project").

Basically, when generating the projects from a BR, there are 4 options:

1. Not generate project (i.e. generated tasks are linked to the master project)
2. Generate 1 project
3. Generate 1 project per DL
4. Generate 1 project per child BR

Ideally, each option should have its own way to store the link with the generated projects.

For now, I suggest the following:

1. Disable option 1. In the future, if we want to re-enable it, we need to store the link with the BR at the task level.
2. Disable option 4. This option is not good enough since it doesn't allow to choose what needs to be generated for the child BRs (e.g. option 2, 3 or 4!)
3. The link between the project and the BR should be stored in the project. This way, we can remove the link stored in the BR and we can also remove the link stored in the tasks.
4. In the project, we should add 2 fields:

- `br_id`: ID of the BR from which the project has been generated
- `dl_id`: only in case of option 3, ID of the DL from which the project has been generated

5. Since we remove the field "Linked Project" in the BR, we need to replace it with a smart button that will give the list of projects that have been generated from this BR.

To be noted that there's a new feature under development to allow having RL directly attached to a BR (without having an intermediate DL). It's the same scenario as option 2.
As for the following scenario where we combine it with option 3, it's possible to implement it this way:
````
BR FOO (id=123)
 > RL 1
 > RL 2
 > RL 3
 > DL A  (id=456)
  > RL A1
  > RL A2
 > DL B  (id=789)
  > RL B1
  > RL B2
  > RL B3
````
If we select options 2 and 3, we would generate 3 projects:
````
FOO (br_id=123, dl_id=NULL)
 > Task 1
 > Task 2
 > Task 3
A (br_id=123, dl_id=456)
 > Task A1
 > Task A2
B (br_id=123, dl_id=789)
 > Task B1
 > Task B2
 > Task B3
````

**Comment by seb-elico** [comment]
@elicoidal following our conversation, I prepared this issue
@victormartinelicocorp can you please prepare a BR to implement my 5 suggestions?

**Comment by elicoidal** [comment]
:+1:

**Comment by elicoidal** [comment]
@seb-elico Let's move fast on this: can you take that up directly with @sudhir-serpentcs?

I think there are multiple areas impacted but all in all those are small changes. Out of my head we need to think about:
* smart buttons in multiple views
* BI report for earned value
* Project generation

**Comment by elicoidal** [comment]
@sudhir-serpentcs could you have a look at this request and whether you could take over?
It is a little bit tricky to understand but the workload should not be that high.
cc @seb-elico

**Comment by seb-elico** [comment]
@sudhir-serpentcs I assigned you task 22070 to implement this request
@victormartinelicocorp please follow-up with Sudhir

**Comment by martintorresvm** [comment]
@YogeshMahera-SerpentCS I mention you as seems you are working on this Issue. Please create or commit on a PR peer day even if it's WIP.

**Comment by elicoidal** [comment]
Fixed by #192
